code/cvm/cosmwasm/contracts/order/src/lib.rs

```diff
@@ -469,18 +505,54 @@ impl OrderContract<'_> {
 			.map(|r| r.map(|(_, x)| x))
 			.collect()
 	}
+
+	/// (partially) fills orders.
+	/// Returns relevant transfers and sets proper tracking info for remaining cross chain
+	/// execution. Orders which are in cross chain execution are "locked", users cannot cancel them
+	/// or take funds back during execution (because funds are moved).
+	fn fill(
+		&self,
+		storage: &mut dyn Storage,
+		cows: Vec<CowFilledOrder>,
+	) -> StdResult<Vec<CowFillResult>> {
+		let mut results = vec![];
+		for (transfer, order) in cows.into_iter() {
```

Comment:
```suggestion
		for (transfer, order) in cows {
```

For loop implicitly calls `into_iter`.